File: src/map_publisher/src/map_publisher.py
#!/usr/bin/env python
import rospy
from nav_msgs.msg import OccupancyGrid
from nav_msgs.msg import MapMetaData
from std_msgs.msg import Header

if __name__ == '__main__':
	rospy.init_node("map_publisher")
	pub = rospy.Publisher("/map",OccupancyGrid, queue_size=10) #I create my publisher
	rate = rospy.Rate(10)
	try:
		rospy.delete_param('/use_sim_time') #Rate.sleep wont work unless you remove this param, the simulator does not appear to publish to /clock
	except:
		print("No parameter to delete")

	mymap = OccupancyGrid( #create my message 2400 pixels wide by 600 pixels long, at 0.05 meters per pixel, AKA 120 meters x 30 meters
		header = Header(seq=0, stamp = rospy.Time.now(), frame_id="map"), 
		info = MapMetaData(width=2400, height=600, resolution=0.05,map_load_time=rospy.Time.now()))

	#As per http://docs.ros.org/jade/api/nav_msgs/html/msg/OccupancyGrid.html
	#The data is in row-major order, starting witht he 0,0 origin.
	#Probabilities are in a range from [0,100], with -1 being 0
	for i in range(0,600*2400): 
		mymap.data.append(-1.0) 

	while not rospy.is_shutdown():
		pub.publish(mymap)
		rate.sleep()

File: src/map_publisher/src/map_publisher.cpp
#include "ros/ros.h"
#include "nav_msgs/OccupancyGrid.h"
#include "nav_msgs/MapMetaData.h"
#include "std_msgs/Header.h"

int main(int argc, char **argv)
{
	ros::init(argc, argv, "talker");
	ros::NodeHandle n;
	ros::Publisher pub = n.advertise<nav_msgs::OccupancyGrid>("/map", 1);

	ros::Rate loop_rate(10);
	
	//Create a header, populate the fields.
	std_msgs::Header header = std_msgs::Header();
	header.stamp = ros::Time::now();
	header.frame_id = "/world";

	//Create the map meta data
	nav_msgs::MapMetaData metaD = nav_msgs::MapMetaData();
	metaD.map_load_time = ros::Time::now();
	metaD.resolution = 0.05; //each pixel will represent .05 meters
	metaD.width = 2400; //2400 pixels long , AKA 120 meters
	metaD.height = 600; //600 pixels tall, AKA 30 meters
	//metaD.origin will just init to 0, no need to change

	nav_msgs::OccupancyGrid map = nav_msgs::OccupancyGrid();
	map.header = header;
	map.info = metaD;
	//As per http://docs.ros.org/jade/api/nav_msgs/html/msg/OccupancyGrid.html
	//The data is in row-major order, starting witht he 0,0 origin.
	//Probabilities are in a range from [0,100], with -1 being 0
	for(int i =0; i < 2400*600; i++)
	{
		map.data.push_back(-1); //Here, I create the entire map as unknown
	}

	while (ros::ok())
	{
		pub.publish(map);
		ros::spinOnce();
	}
}
